File: src/c/ch01/1.2/l2/exercises_starter.c
#include <stdio.h>
#include "karatsuba.h"
#include "modexp.h"

void exercise_1(void) {
    long long res = karatsuba(31415926LL, 27182818LL);
    printf("Exercise 1 (Karatsuba 31415926*27182818): %lld\n", res);
}

void exercise_2(void) {
    long long res = modexp(5,117,19);
    printf("Exercise 2 (modexp 5^117 mod 19): %lld\n", res);
}

void exercise_3(void) {
    long long a = 12345, b = 67890;
    long long res = karatsuba(a,b);
    printf("Exercise 3 (check Karatsuba correctness): %s\n", 
           (res == a*b) ? "true" : "false");
}

int main(void) {
    exercise_1();
    exercise_2();
    exercise_3();
    return 0;
}

File: src/go/ch01/1.2/l2/exercises_starter.go
package main

import "fmt"

func ex1Karatsuba() int {
	return Karatsuba(31415926, 27182818)
}

func ex2ModExp() int {
	return ModExp(5, 117, 19)
}

func ex3Check() bool {
	a, b := 12345, 67890
	return Karatsuba(a, b) == a*b
}

func runExercises() {
	fmt.Println("Exercise 1 (Karatsuba 31415926*27182818):", ex1Karatsuba())
	fmt.Println("Exercise 2 (modexp 5^117 mod 19):", ex2ModExp())
	fmt.Println("Exercise 3 (check Karatsuba correctness):", ex3Check())
}
